lib/python/rose/apps/fcm_make.py

```diff
@@ -47,91 +54,192 @@ def run(self, app_runner, conf_tree, opts, args, uuid, work_files):
         This application will only work under "rose task-run".
 
         """
-        orig_cont_map = conf_tree.node.get_value(
-            ["orig-cont-map"], ":".join(self.ORIG_CONT_MAP)).split(":", 1)
+        # Determine if this is an original task or a continuation task
+        orig_cont_map = _conf_value(
+            conf_tree, ["orig-cont-map"], ":".join(self.ORIG_CONT_MAP)
+        ).split(":", 1)
         task = app_runner.suite_engine_proc.get_task_props()
 
         if orig_cont_map[CONT] in task.task_name:
             return self._run_cont(
-                app_runner, conf_tree, args, task, orig_cont_map)
+                app_runner, conf_tree, opts, args, uuid, task, orig_cont_map)
+        else:
+            return self._run_orig(
+                app_runner, conf_tree, opts, args, uuid, task, orig_cont_map)
 
+    def _get_fcm_make_cmd(self, conf_tree, opts, args, dest, make_name):
+        """Return a list containing the "fcm make" command to invoke."""
         cmd = ["fcm", "make"]
-        use_pwd = conf_tree.node.get_value(["use-pwd"]) in ["True", "true"]
-        for cfg in [
-                "fcm-make.cfg",
-                os.path.join(task.suite_dir, "etc", task.task_name + ".cfg")]:
-            if os.access(cfg, os.F_OK | os.R_OK):
-                if cfg != "fcm-make.cfg" or not use_pwd:
-                    cmd += ["-f", os.path.abspath(cfg)]
-                break
-        if not use_pwd:
-            cmd += [
-                "-C",
-                os.path.join(task.suite_dir, "share", task.task_name)]
-        cmd += ["-j", conf_tree.node.get_value(
-            ["opt.jobs"], os.getenv("ROSE_TASK_N_JOBS", self.OPT_JOBS))]
-        cmd_args = conf_tree.node.get_value(
-            ["args"], os.getenv("ROSE_TASK_OPTIONS"))
+        if make_name is None:
+            make_name = ""
+        cfg_file_name = self.CFG_FILE_NAME % {"name": make_name}
+        if os.access(cfg_file_name, os.F_OK | os.R_OK) and dest:
+            cmd += ["-f", os.path.abspath(cfg_file_name)]
+        if dest:
+            cmd += ["-C", dest]
+        if make_name:
+            # "-n NAME" option requires fcm-2015.05+
+            cmd += ["-n", make_name]
+        if opts.new_mode:
```

Comment:
Possibly, but not sure if it should be part of this change or not. (However, it is worth noting that from now on, it will be much better to use `rose task-run --new` to impose new mode.)